### PR TITLE
gh-131: rename `gaussian_gls` to `discretized_cls`

### PR DESCRIPTION
**Pull request by Saransh-cpp**
This popped up while I was looking to add tests for `fields.py`.

- rename `gaussian_gls` to `discretized_cls`
- update docstring and internal references
- don't use `gaussian_gls` in `lognormal_gls`

Closes: #131
Changed: `gaussian_gls` has been renamed to `discretized_cls`
Changed: `lognormal_gls` no longer uses `gaussian_gls` internally and the arguments passed to `gaussian_gls` have been removed

### CODE REVIEW

File: examples/1-basic/cls.npy
Comment (paddyroddy) [inline examples/1-basic/cls.npy:1]: This should be in `.gitignore`

**Review by paddyroddy — CHANGES_REQUESTED** [review]
Looks good to me, although @ntessore should definitely have the final say

**Comment by ntessore** [comment]
One critical thing that is missing here is that, by taking `discretized_cls()` out of `lognormal_gls()`, the discretisation is no longer applied to the `cls` used in the examples.

I think the "discretisation step" would be best shown in each example. Something along the lines of:

```diff
-# compute Gaussian cls for lognormal fields for 3 correlated shells
-gls = glass.lognormal_gls(cls, nside=nside, lmax=lmax, ncorr=3)
+# apply discretisation to the full set of spectra:
+# - HEALPix pixel window function (`nside=nside`)
+# - maximum angular mode number (`lmax=lmax`)
+# - number of correlated shells (`ncorr=3`)
+cls = glass.discretized_cls(cls, nside=nside, lmax=lmax, ncorr=3)
+
+# compute Gaussian cls for lognormal fields for 3 correlated shells
+gls = glass.lognormal_gls(cls)
```

As a matter of fact, since notebooks are run locally and their outputs stored (i.e., `cls.npz`), it probably no longer makes sense to pre-compute the Cls for other examples; that was only done to save time when running on GitHub actions.

**Comment by Saransh-cpp** [comment]
Thanks for the help! I have updated the notebooks to show the discretization step.

File: examples/1-basic/shells.ipynb
Comment (ntessore) [inline examples/1-basic/shells.ipynb:1]: I wouldn't remove this example entirely, as it's still useful as a first step towards understanding how the simulations are set up. In fact, I would incorporate #334 here in the future.

**Comment by ntessore** [comment]
Thanks for the great work @Saransh-cpp, and sorry for the annoying change requests. Here is one more: The `## Setup` section still mentions that the examples use pre-computed spectra. (The text saying that the set up matches `shells` can stay and is probably useful to guide newcomers to a more basic example.)

**Comment by Saransh-cpp** [comment]
No worries, @ntessore! I have updated the PR. Could you please take a look again?

**Comment by ntessore** [comment]
Unrelated to this PR, but we should tweak the "Test examples" action to not just run the examples, but also produce and upload HTML artefacts of the results.

**Comment by Saransh-cpp** [comment]
Thanks! I have created relevant issues for the comments above.

File: examples/1-basic/shells.ipynb
Comment (paddyroddy) [reply examples/1-basic/shells.ipynb:1]: Not sure if this is a GitHub rendering issue. I can't see the changes
image

Comment (ntessore) [reply examples/1-basic/shells.ipynb:1]: Only in "Files changed" for me too when it comes to notebooks!

Comment (ntessore) [reply examples/1-basic/shells.ipynb:1]: Same, but I can see the full file w/ changes under "View file"

Comment (ntessore) [reply examples/1-basic/shells.ipynb:1]: And the source diff!

**Review by ntessore — APPROVED** [review]

**Review by paddyroddy — APPROVED** [review]

**Pull request merged**